### PR TITLE
Ubuntu 20.04 updates

### PR DESCRIPTION
**Pull request by michael-balint**
* migrated from python2 to python3

Tested on Ubuntu 20.04:
- [x] nvidia-dgx
- [x] k8s-cluster
- [x] slurm-cluster

Tested on Ubuntu 18.04:
- [x] nvidia-dgx
- [x] k8s-cluster
- [x] slurm-cluster

### CODE REVIEW

**Comment by dholt** [comment]
Made a couple small PRs for 20.04 but otherwise the Slurm cluster stuff looks good to me

File: roles/slurm/vars/ubuntu.yml [hunk 1]
```diff
@@ -4,10 +4,10 @@ slurm_pam_lib_dir: /lib/x86_64-linux-gnu/security
 slurm_build_deps:
   - build-essential
   - libmunge-dev
-  - libmariadb-client-lgpl-dev
-  - libmysqlclient-dev
```
Comment (dholt) [inline roles/slurm/vars/ubuntu.yml:8]: I believe removing these packages is causing `slurmdbd` to fail on Ubuntu 18.04:

```
[2020-10-16T16:57:56.621] error: Couldn't find the specified plugin name for accounting_storage/mysql looking at all files
[2020-10-16T16:57:56.622] error: cannot find accounting_storage plugin for accounting_storage/mysql
[2020-10-16T16:57:56.622] error: cannot create accounting_storage context for accounting_storage/mysql
[2020-10-16T16:57:56.622] fatal: Unable to initialize accounting_storage/mysql accounting storage plugin
```

Comment (dholt) [reply roles/slurm/vars/ubuntu.yml:8]: Replacing `libmariadb-client-lgpl-dev` with `libmariadb-dev` for both Ubuntu 18.04 and 20.04 is OK. It looks like on 18.04 we should install `libmysqlclient-dev` and on 20.04 we should install `libmariadb-dev-compat`.

Comment (dholt) [reply roles/slurm/vars/ubuntu.yml:8]: Can confirm this works on both versions:

```
  - libmariadb-dev
  - libmariadbclient-dev
```

File: roles/nvidia-dgx/tasks/ubuntu.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 ---
+- name: Check for Ubuntu version
+  fail:
+    msg: "Role supports Ubuntu 18.04 only"
+  when: ansible_distribution_version != "18.04"
+
```
Comment (dholt) [inline roles/nvidia-dgx/tasks/ubuntu.yml:6]: Just a suggestion for future-proofing: we may want to re-name the `ubuntu.yml` file to something like `ubuntu-18.04.yml` and put this check in `main.yml`, i.e.

```
- name: ubuntu tasks
  include_tasks: ubuntu-18.04.yml
  when:
    - ansible_distribution == 'Ubuntu'
    - ansible_distribution_version == "18.04"
```

Comment (michael-balint) [reply roles/nvidia-dgx/tasks/ubuntu.yml:6]: Done!

**Review by dholt — APPROVED** [review]

**Pull request merged**